### ISSUE TITLE
TwoStageFallbackActions not Working 

### ISSUE THREAD

**Issue by tsturdivant** [RasaHQ/rasa#5195]
**Rasa version**: 1.6.1

**Rasa SDK version** (if used & relevant): 1.6.0

**Rasa X version** (if used & relevant): not used 

**Python version**: 3.6.8

**Operating system** (windows, osx, ...): osx

**Issue**:
When I try to use two stage fallback policy, denying will lead to the bot getting stuck in a loop where it doesn't understand intents that it would understand before. 
Working Output:
```
Your input ->  hello                                               
Hey! How are you?
Your input ->  good                                         
Great, carry on!
```
Getting Stuck:
```
Your input ->  hello                                               
Hey! How are you?
Your input ->  gud                                                 
? Did you mean 'affirm'?  2: No (/out_of_scope)                    
Could you rephrase this please.
Your input ->  gud                                                 
? Did you mean 'affirm'?  2: No (/out_of_scope)                    
Could you rephrase your message? I didn't get it, I'm sorry.
Your input ->  good                                                
I'm afraid I didn't get what you just said. Could you rephrase that?
Your input ->  good                                 
I'm afraid I didn't get what you just said. Could you rephrase that?
```

As you can see after denying the intent twice it no longer understands when I type good, even though it understands perfectly before that. 

**Command or request that led to error**:
```
rasa shell
```

**Content of configuration file (config.yml)** (if relevant):
```yml
# Configuration for Rasa NLU.
# https://rasa.com/docs/rasa/nlu/components/
language: en
pipeline: 
  - name: "SpacyNLP"
  - name: "SpacyTokenizer"
  - name: "SpacyFeaturizer"
    return_sequence: true
  - name: "RegexFeaturizer"
    return_sequence: true
  - name: "CRFEntityExtractor"
  - name: "EntitySynonymMapper"
  - name: "SklearnIntentClassifier"
  - name: "ResponseSelector"


# Configuration for Rasa Core.
# https://rasa.com/docs/rasa/core/policies/
policies:
  - name: MemoizationPolicy
  - name: KerasPolicy
  - name: MappingPolicy
  - name: TwoStageFallbackPolicy
    nlu_threshold: 0.6
    core_threshold: 0.3

```

**Content of domain file (domain.yml)** (if relevant):
```yml
actions:
- action_session_start
- respond_faq
- respond_out_of_scope
- utter_cheer_up
- utter_default
- utter_did_that_help
- utter_goodbye
- utter_greet
- utter_happy
- utter_iamabot
- utter_restart_with_button
- utter_sign_up
entities:
- question_regards
- question_subject
intents:
- goodbye
- affirm
- session_start
- greet
- mood_unhappy
- stop
- top_faqs
- stop_shop_card
- request_stop_and_shop_card:
    use_entities: []
- deny
- mood_great
- bot_challenge
- out_of_scope
session_config:
  carry_over_slots_to_new_session: true
  session_expiration_time: 60
slots:
  feedback_value:
    type: unfeaturized
  question_regards:
    type: unfeaturized
  question_subject:
    type: unfeaturized
templates:
  respond_out_of_scope:
  - text: I'm sorry. That is currently outside of my capabilities.
  utter_cheer_up:
  - image: https://i.imgur.com/nGF1K8f.jpg
    text: 'Here is something to cheer you up:'
  utter_default:
  - text: Sorry, I didn't get that 🤷. Could you please rephrase?
  - text: I didn't understand, could you rephrase that?
  - text: I'm sorry, but I didn't understand you. Could you please rephrase what you
      just said?
  - text: I'm afraid I didn't get what you just said. Could you rephrase that?
  - text: I didn't quite get that, could you rephrase your message?
  - text: Could you rephrase your message? I didn't get it, I'm sorry.
  utter_did_that_help:
  - text: Did that help you?
  utter_goodbye:
  - text: Bye
  utter_greet:
  - text: Hey! How are you?
  utter_happy:
  - text: Great, carry on!
  utter_iamabot:
  - text: "Hi! I'm a bot with no name. Do not call me Betty.\nWelcome to Stop & Shop!\
      \ Looking for quick customer service? I can help you with\nany queries you might\
      \ have about Stop & Shop or the Stop & Shop Card."
  utter_restart_with_button:
  - buttons:
    - payload: /restart
      title: Restart
    text: Click the button below if you want to start over.
  utter_sign_up:
  - text: HERE IS WHERE THE FORM WILL BE FILLED OUT
  utter_ask_rephrase:
  - text: "Could you rephrase this please."
```

**Comment by sara-tagger** [comment]
Thanks for the issue, @amn41 will get back to you about it soon!

###### You may find help in the [docs](https://rasa.com/docs/) and the [forum,](https://forum.rasa.com/) too 🤗

**Comment by tsturdivant** [comment]
After spending some time to play around with this I have found a solution for this. 
In config.yml I have added the fallback_nlu_action_name to the policy:
```yml
policies:
  - name: MemoizationPolicy
  - name: KerasPolicy
  - name: MappingPolicy
  - name: TwoStageFallbackPolicy
    nlu_threshold: 0.6
    core_threshold: 0.3
    fallback_nlu_action_name: "action_default_fallback"
```
And in my actions.py I have added code for my own ultimate fallback: 
```python
class ActionDefaultFallback(Action): 
    def name(self):
        return 'action_default_fallback'

    
    def run(self, dispatcher, tracker, domain): 
        return dispatcher.utter_message(text="I'm unable to process your request.\nIf you would like to speak with a human you can contact them with this number:\n(555)555-5555")
```

This is allowing me to trigger the ultimate fallback whenever I deny twice in a row. However if I am understanding correctly the default TwoStageFallback policy should work without the need of me writing a custom action. I'm not sure if this has been glossed over in the documentation that writing the policy is necessary for this to work or if I simply missed it when reading, but I feel that if I want to use the default version I should not have to write my own custom action.
